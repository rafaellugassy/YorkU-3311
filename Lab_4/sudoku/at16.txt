set_number(3, 2, 1) 	-- 1
set_number(2, 2, 2) 	-- 2

start_game 				-- 3

put_number(1, 3, 3) 	-- 4
put_number(3, 3, 4) 	-- 5

restart_game			-- 6

put_number(4, 2, 3)		-- error: game already started (so not included in history)

set_number(4, 2, 3)		-- 7

undo -- back to after executing 6
undo -- error: cannot undo beyond the last restart_game

redo -- back to after executing 7
redo -- nothing to redo
