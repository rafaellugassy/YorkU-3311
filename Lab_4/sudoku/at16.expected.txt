  Setting up a new grid
  0   0   0   0
  0   0   0   0
  0   0   0   0
  0   0   0   0
->set_number(3,2,1)
  Setting up a new grid
  0   0   0   0
  3   0   0   0
  0   0   0   0
  0   0   0   0
->set_number(2,2,2)
  Setting up a new grid
  0   0   0   0
  3   2   0   0
  0   0   0   0
  0   0   0   0
->start_game
  Current game is over: No
  0   0   0   0
  3   2   0   0
  0   0   0   0
  0   0   0   0
->put_number(1,3,3)
  Current game is over: No
  0   0   0   0
  3   2   0   0
  0   0   1   0
  0   0   0   0
->put_number(3,3,4)
  Current game is over: No
  0   0   0   0
  3   2   0   0
  0   0   1   3
  0   0   0   0
->restart_game
  Setting up a new grid
  0   0   0   0
  0   0   0   0
  0   0   0   0
  0   0   0   0
->put_number(4,2,3)
  Error: Game not yet started
  0   0   0   0
  0   0   0   0
  0   0   0   0
  0   0   0   0
->set_number(4,2,3)
  Setting up a new grid
  0   0   0   0
  0   0   4   0
  0   0   0   0
  0   0   0   0
->undo
  Setting up a new grid
  0   0   0   0
  0   0   0   0
  0   0   0   0
  0   0   0   0
->undo
  Error: Nothing to undo
  0   0   0   0
  0   0   0   0
  0   0   0   0
  0   0   0   0
->redo
  Setting up a new grid
  0   0   0   0
  0   0   4   0
  0   0   0   0
  0   0   0   0
->redo
  Error: Nothing to redo
  0   0   0   0
  0   0   4   0
  0   0   0   0
  0   0   0   0
